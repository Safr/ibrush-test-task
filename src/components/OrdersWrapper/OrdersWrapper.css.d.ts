export const OrdersWrapper: string;
export const ordersWrapper: string;
export const TraiderHeader: string;
export const traiderHeader: string;
export const CurrencyHeader: string;
export const currencyHeader: string;
