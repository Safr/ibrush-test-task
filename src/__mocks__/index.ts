export const orderBook = {
  BTC_USD: {
    ask: ['6891', '0.005', '34.455'],
    ask_amount: '2413543.16939829',
    ask_quantity: '190.96765367',
    ask_top: '6887.20157116',
    bid: ['6849.7394927', '0.75', '5137.30461952'],
    bid_amount: '286086.48310063',
    bid_quantity: '11524.76654836',
    bid_top: '6875',
  },
};

export const order = {
  avg: '6875.86448007',
  buy_price: '6859.227881',
  high: '6969.54201841',
  last_trade: '6864.01',
  low: '6800',
  sell_price: '6873.71642794',
  updated: 1528752429,
  vol: '1112.01492053',
  vol_curr: '7632881.53471028',
};

export const currentOrder = {
  BTC_USD: {
    avg: '6875.86448007',
    buy_price: '6859.227881',
    high: '6969.54201841',
    last_trade: '6864.01',
    low: '6800',
    sell_price: '6873.71642794',
    updated: 1528752429,
    vol: '1112.01492053',
    vol_curr: '7632881.53471028',
  },
};

export const orders = { currentOrder };
export const changedOrders = [''];
export const orderBookInfo = [['5897.8242', '0.003', '17.6934726']];

// export const fetchOrders = () => Promise.resolve({
//   orders: {
//     BTC_USD: {
//       avg: '6875.86448007',
//       buy_price: '6859.227881',
//       high: '6969.54201841',
//       last_trade: '6864.01',
//       low: '6800',
//       sell_price: '6873.71642794',
//       updated: 1528752429,
//       vol: '1112.01492053',
//       vol_curr: '7632881.53471028',
//     },
//   },
// });
