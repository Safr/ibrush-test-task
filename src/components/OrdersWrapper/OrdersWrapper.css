.OrdersWrapper {
  padding: 20px;
  width: 90%;
  margin-bottom: 100px;
  text-align: center;
  background-color: #bbe5f1;
  box-shadow: 20px 38px 34px -26px hsla(0,0%,0%,.2);
  border-radius: 255px 15px 225px 15px/15px 225px 15px 255px;
  border:solid 2px #41403E;
}

@media screen and (max-width: 660px) {
  .OrdersWrapper {
    width: 98%;
    margin-bottom: 50px;
  }
}

.OrdersWrapper:hover {
  box-shadow:2px 8px 4px -6px hsla(0,0%,0%,.3);
}

.TraiderHeader {
  margin-top: 0;
  margin-bottom: 30px;
  text-align: left;
}

.CurrencyHeader {
  margin: 0;
  margin-bottom: 20px;
  text-align: left;
  font-weight: normal;
}