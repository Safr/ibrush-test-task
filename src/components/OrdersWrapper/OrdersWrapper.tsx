import React from 'react';
import classes from './OrdersWrapper.css';
import OrdersList from '../OrdersList/OrdersList';
import Spinner from '../Spinner/Spinner';
import { OrderInterface } from '../../types';

interface Props {
  orders: null | object;
  changedOrders: string[];
  isFetching: boolean;
  onOpenPopup: (orderName: string, className: string, order: OrderInterface) => void;
  error: string;
}

const OrdersWrapper: React.SFC<Props> = ({ orders, changedOrders, isFetching, onOpenPopup, error }) => {
  return (
    <div className={classes.OrdersWrapper}>
      <h2 className={classes.TraiderHeader}>Сам себе трейдер</h2>
      <h3 className={classes.CurrencyHeader}>Валютные пары</h3>
      {error ? <h3>{error}</h3> : null}
      {isFetching && <Spinner />}
      {!isFetching && orders &&
        <OrdersList orders={orders} changedOrders={changedOrders} onOpenPopup={onOpenPopup} />}
    </div>
  );
};

export default OrdersWrapper;
