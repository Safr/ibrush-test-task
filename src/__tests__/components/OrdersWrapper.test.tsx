import React from 'react';
import { shallow } from 'enzyme';
import OrdersWrapper from '../../components/OrdersWrapper/OrdersWrapper';
import { orders, changedOrders } from '../../__mocks__';

describe('OrdersWrapper Component', () => {
  test('should render correctly', () => {
    const wrapper = shallow(
      // tslint:disable-next-line:jsx-no-lambda
      <OrdersWrapper
        orders={orders}
        changedOrders={changedOrders}
        isFetching={false}
        // tslint:disable-next-line:jsx-no-lambda
        onOpenPopup={() => null}
        error={''}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
