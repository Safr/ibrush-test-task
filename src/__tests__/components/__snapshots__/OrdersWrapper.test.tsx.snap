// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrdersWrapper Component should render correctly 1`] = `
<div>
  <h2>
    Сам себе трейдер
  </h2>
  <h3>
    Валютные пары
  </h3>
  <OrdersList
    changedOrders={
      Array [
        "",
      ]
    }
    onOpenPopup={[Function]}
    orders={
      Object {
        "currentOrder": Object {
          "BTC_USD": Object {
            "avg": "6875.86448007",
            "buy_price": "6859.227881",
            "high": "6969.54201841",
            "last_trade": "6864.01",
            "low": "6800",
            "sell_price": "6873.71642794",
            "updated": 1528752429,
            "vol": "1112.01492053",
            "vol_curr": "7632881.53471028",
          },
        },
      }
    }
  />
</div>
`;
